yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1284,6 +1284,191 @@ def pixelize_sph_kernel_projection(
 
     return mask
 
+@cython.initializedcheck(False)
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def pixelize_sph_kernel_projection_with_depth_check(
+        np.float64_t[:, :] buff,
+        np.uint8_t[:, :] mask,
+        any_float[:] posx,
+        any_float[:] posy,
+        any_float[:] posz,
+        any_float[:] hsml,
+        any_float[:] pmass,
+        any_float[:] pdens,
+        any_float[:] quantity_to_smooth,
+        bounds,
+        kernel_name="cubic",
+        weight_field=None,
+        int check_period=1,
+        period=None):
+
+    cdef np.intp_t xsize, ysize
+    cdef np.float64_t x_min, x_max, y_min, y_max, z_min, z_max, prefactor_j
+    cdef np.int64_t xi, yi, x0, x1, y0, y1, xxi, yyi
+    cdef np.float64_t q_ij2, posx_diff, posy_diff, ih_j2
+    cdef np.float64_t x, y, dx, dy, idx, idy, h_j2, px, py, pz
+    cdef np.float64_t period_x = 0, period_y = 0, period_z = 0
+    cdef int i, j, ii, jj, kk
+    cdef np.float64_t[:] _weight_field
+    cdef int * xiter
+    cdef int * yiter
+    cdef int * ziter
+    cdef np.float64_t * xiterv
+    cdef np.float64_t * yiterv
+    cdef np.float64_t * ziterv
+
+    if weight_field is not None:
```

Comment:
I think you got this code from elsewhere, but I will say that it definitely confused me.  I at first thought this was a string, but now I see it's either a numpy array or none.  Yowza!

Reply:
indeed! copied directly from `pixelize_sph_kernel_projection`. Up in the python level there's a `weight` that is a string, so definitely confusing. Could rename this variable to `weight_array` (and also rename in `pixelize_sph_kernel_projection`).